src/en/news/blog/2021/managing-grafana-dashboards-with-grafonnet/index.md

```diff
@@ -0,0 +1,94 @@
+The code below is the minimal code required to create a dashboard requiring only a title.
+
+```
+local grafana = import 'grafonnet/grafana.libsonnet';
+local dashboard = grafana.dashboard;
+ 
+dashboard.new(
+  title='Demo dashboard'
+)
+```
+
+The above code can be compiled with `$ jsonnet -o new.json new.jsonnet` and the resultant json file(new.json) can be imported into Grafana.
+
+## Adding Template Variables
+
+Grafonnet library provides the functions `addTemplate() and addTemplates()`, which either take a template object or an array of them.
+
+```
+local template = grafana.template;
+ 
+dashboard.new(
+  title='Demo Dashboard'
+)
+.addTemplate(
+  template.new(
+    name='service',
+    datasource='LocalPrometheus',
+    query='label_values(service)',
+    allValues=null,
+    current='all',
+    refresh='load',
+    includeAll=true,
+    multi=true,
+  )
+)
+```
+
+## Adding Panels
+
+Panel objects are being added similar to template variables, by using `addPanel() or addPanels()`. In addition to creating the panel objects, we specify their position by adding a `gridPos` attribute. Below is the additional code to above for adding one text and graph panel.
+By using `addTarget()` which is exposed by the panel object we can add a target object.
+
+```
+local prometheus = grafana.prometheus;
+ 
+// dashboard and template objects
+.addPanels([
+  textPanel.new(
+    mode='html',
+    content='<p2 align=\"center\">Services</p2>',
+  ) + {gridPos: {h: 1, w: 0, x: 8, y: 0}},
+  graphPanel.new(
+    title='RGW Sync Overview',
+    fill=2,
+    formatY1='percentunit',
+    datasource='Prometheus',
+  )
+  .addTargets([
+    prometheus.target(
+      expr='ceph_health_status == 2',
+      intervalFactor=1,
+      legendFormat='time_series',
+    )
+  ]) + {gridPos: {h: 0, w: 0, x: 8, y: 9}},
+])
+```
+
+## Custom functions for Reusability
+
+Using reusable functions will make the code not just easier to read but easier to maintain as well.Below is an example of making custom functions which take properties such as title, uid etc. and these functions can be reused if you want to create a number of dashboards by just passing the parameters.
+
+```
+local dashboardSchema(title, uid, time_from, refresh, schemaVersion, tags,timezone, timepicker) =
+  g.dashboard.new(title=title, uid=uid, time_from=time_from, refresh=refresh, schemaVersion=schemaVersion, tags=tags, timezone=timezone, timepicker=timepicker);
+
+local graphPanelSchema(title, nullPointMode, stack, formatY1, formatY2, labelY1, labelY2, min, fill, datasource) =
+  g.graphPanel.new(title=title, nullPointMode=nullPointMode, stack=stack, formatY1=formatY1, formatY2=formatY2, labelY1=labelY1, labelY2=labelY2, min=min, fill=fill, datasource=datasource);
+
+```
```

Comment:
From an engineering standpoint, I'd add a "Results" section: something that is relevant for this move is the number of lines of code to maintain before/after this change. So you can add some charts (or just a stat): from (fictional figures) 2000 lines of code to 100, hence a 95% reduction. You have already mentioned reusability, etc...

I'd also suggest to add a "migration" section, where you share your tips for migrating from existing JSON dashboards.

Additionally, you can also mention the tooling ecosystem (jsonnet-lint, etc.) and your future plans for improving this even more (maybe unit testing Grafana PromQL `expr`).